File: sdk/typescript/examples/quickstart.ts
// Minimal SuperBased Observer TypeScript SDK example.
//
//   cd sdk/typescript && npm install && npm run build
//   node --loader ts-node/esm examples/quickstart.ts   # or compile + run
//
// Requires a local Observer with [observability] enabled.

import { init, Kind, setUsage, shutdown, withLlmSpan, withSpan } from "../src/index.js";

async function main() {
  init({ sessionId: "quickstart-1", user: "demo" });

  await withSpan("agent.run", Kind.AGENT, async () => {
    await withLlmSpan("chat", { model: "gpt-4o", provider: "openai" }, async (span) => {
      // ... your real model call ...
      setUsage(span, { inputTokens: 1200, outputTokens: 64, responseId: "chatcmpl-demo-123" });
    });
    await withSpan("web_search", Kind.TOOL, async () => {
      // ... your tool call ...
    });
  });

  await shutdown();
  console.log("sent one trace to Observer — check /trajectories");
}

void main();

File: sdk/python/examples/quickstart.py
"""Minimal SuperBased Observer SDK example.

Run a local Observer with [observability] enabled, then:

    pip install -e sdk/python
    python sdk/python/examples/quickstart.py

It emits one trace (an agent span containing an LLM span and a tool span) to
the OTLP receiver at http://127.0.0.1:4318/v1/traces. Open the dashboard's
Trajectories page to see it.
"""

import superbased


def main() -> None:
    superbased.init(session_id="quickstart-1", user="demo")

    with superbased.span("agent.run", kind=superbased.AGENT):
        with superbased.llm_span("chat", model="gpt-4o", provider="openai") as s:
            # ... your real model call goes here ...
            superbased.set_usage(
                s,
                input_tokens=1200,
                output_tokens=64,
                response_id="chatcmpl-demo-123",
            )
        with superbased.span("web_search", kind=superbased.TOOL):
            pass  # ... your tool call ...

    superbased.shutdown()  # flush before exit
    print("sent one trace to Observer — check /trajectories")


if __name__ == "__main__":
    main()
